components/org.wso2.carbon.identity.oauth.common/src/test/java/org/wso2/carbon/identity/oauth/common/CodeTokenResponseValidatorTest.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.oauth.common;
+
+import org.apache.oltu.oauth2.common.error.OAuthError;
+import org.apache.oltu.oauth2.common.exception.OAuthProblemException;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
+
+import static org.powermock.api.mockito.PowerMockito.mock;
+import static org.powermock.api.mockito.PowerMockito.when;
+import static org.testng.Assert.assertTrue;
+import static org.testng.Assert.fail;
+import static org.wso2.carbon.identity.oauth.common.OAuthConstants.OAuth20Params.NONCE;
+import static org.wso2.carbon.identity.oauth.common.OAuthConstants.OAuth20Params.SCOPE;
+
+/**
+ * Test class for CodeTokenResponse Validator.
+ */
+public class CodeTokenResponseValidatorTest {
+
+    protected CodeTokenResponseValidator testedResponseValidator;
+
+    @BeforeMethod
+    public void setUp() throws Exception {
+
+        testedResponseValidator = new CodeTokenResponseValidator();
+    }
+
+    @AfterMethod
+    public void tearDown() throws Exception {
+    }
+
+    @DataProvider(name = "Request Provider")
+    public Object[][] getRequestParams() {
+
+        Map<String, String> allParamPresentMap = new HashMap<>();
+        allParamPresentMap.put(NONCE, "nonce");
+        allParamPresentMap.put(SCOPE, OAuthConstants.Scope.OPENID);
+        Map<String, String> nonOIDCScopeMap = new HashMap<>();
+        nonOIDCScopeMap.put(NONCE, "nonce");
+        nonOIDCScopeMap.put(SCOPE, "notOpenid");
+        Map<String, String> blankNonceMap = new HashMap<>();
+        blankNonceMap.put(NONCE, "");
+        blankNonceMap.put(SCOPE, "notOpenid");
+        Map<String, String> blankScopeMap = new HashMap<>();
+        blankScopeMap.put(NONCE, "nonce");
+        blankScopeMap.put(SCOPE, "");
```

Comment:
Don't we need to check both NONCE, SCOPE empty? and wrong nonce value, wrong value for openid?